src/OpenColorIO/Platform.cpp

```diff
@@ -180,20 +168,21 @@ std::string CreateTempFilename(const std::string & filenameExt)
         throw Exception("Could not create a temporary file.");
     }
 
-    filename = tmpFilename;
+    filename = tmpFilename[0] == '\\' ? tmpFilename + 1 : tmpFilename;
+    filename += filenameExt;
 
 #else
 
     // Linux flavors must have a /tmp directory.
-    std::stringstream ss;
-    ss << "/tmp/ocio_" << GenerateRandomNumber();
-
-    filename = ss.str();
+    filename = "/tmp/ocio_XXXXXX" + filenameExt;
+    
+    if(mkstemps(filename.data(), filenameExt.size()) == -1)
```

Comment:
The change is major as the method `mkstemps` now generates a temporary file name, creates the file on disk, opens it and returns the file descriptor. I mainly have two concerns:
1. The implementation behavior is now different between `Linux` and `Windows` which is confusing.
2. The file creation & opening is clearly not always needed. You can check `OCIO_ADD_TEST(Platform, create_temp_filename)` and  `OCIO_ADD_TEST(CDLTransform, escape_xml)` as example where these steps are useless.

Reply:
The implementation could be made the same on both platforms. What I'd do is make it return an std::fstream. I think it would be good to also test creation and opening of these files. It could also be more secure because mkstemps will assign permission of 0600 on the file. It will also open the file with O_EXCL flag so opening will fail if it already exists, safer and one less thing to think about. The only trouble is to pass that file descriptor to a fstream, there is no standard way.

In any case, if it's too much of a change, then nevermind.

Reply:
Looks like we could have `CreateTempFilename` left as is and add `CreateTempFile` next to it ? Agree that the Windows path need to create a file as well.

Reply:
Here is a summary of the requirements:
- The behavior must be exactly the same between the different platforms.
- The only usages is to create a temporary file with a predefined content (it implies to later automatically delete the file).
   - The only usages are currently in CPU unit tests for `CDL` tests and in the GPU unit tests for 3D LUT tests. (The test in `Platform_tests.cpp` is not a usage because it tests the method itself).
- To use the method in the GPU unit test, the method must be public but the std::stream's are strictly forbidden in the public API (in fact all std classes/methods are forbidden)

So, I can only see two approaches:
1. _Public method:_ Improve the existing method (it only creates a filename) and make it public (to refactor the GPU method `createTempFile()`).
2. _Public class:_ Create a public `TempFile` class which creates a temporary file with a predefined content and automatically deletes the temporary file in the destructor (a little bit like the `FileGuard` in CLDTransform_tests.cpp).

Note:
> Looks like we could have CreateTempFilename left as is and add CreateTempFile next to it ?

In my mind that's similar to the class proposal because the file temporary name is only needed to create a file, and the temporary file deletion must still be consider.